.github/workflows/pr-auto-merge.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  pull_request:
+
+jobs:
+  e2e:
+    uses: ./.github/workflows/_e2e.yaml
+
+  merge_pr_when_e2e_success:
+    name: Merge PR when E2E success
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    needs:
+      - e2e
+    if: ${{ always() && needs.e2e.result == 'success' && contains('["dependabot","github-actions"]', github.event.pull_request.user.login) }}
+    steps:
+      - name: Merge PR
+        run: |
+          gh pr merge --merge --auto "${{github.event.pull_request.html_url}}"
+
+  notify_slack_when_e2e_failed:
+    name: Notify Slack when E2E failed
+    runs-on: ubuntu-latest
+    needs:
+      - e2e
+    if: ${{ always() && needs.e2e.result == 'failure' && contains('["dependabot","github-actions"]', github.event.pull_request.user.login) }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/notify-slack
+        with:
+          author_name: E2E failed on PR
+          slack_notify_webhook_url: ${{ secrets.SLACK_WEBHOOK_URL }}
+          status: failure
```

Comment:
Is it better to implement `_e2e.yaml` instead of this workflow?

Reply:
related here:
https://github.com/vdaas/vald-client-java/pull/180/files#r1296615745